File: bloc_de_horarios.java
import java.util.Scanner;
public class bloc_de_horarios {
    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);
        int opcion = 0;
        int hora = 0;
        String actividad = "";
        String nota = "";
        int total_actividades = 0;
        int actividades_completadas = 0;
        int actividades_pendientes = 0;
        String[] actividades = new String[24];
        String[] notas = new String[24];
        boolean[] completadas = new boolean[24];
        for (int i = 0; i < 24; i++) {
            actividades[i] = "";
            notas[i] = "";
            completadas[i] = false;
        }
        //-------------------------------------//
        //--|menu_principal_bloc_de_horarios|--//
        //-------------------------------------//
        while (true) {
            System.out.println("=================================");
            System.out.println("        BLOC DE HORARIOS");
            System.out.println("=================================");
            System.out.println("1. Ver horario por dia");
            System.out.println("2. Ver horario por semana");
            System.out.println("3. Ver horario por mes");
            System.out.println("4. Registrar actividad");
            System.out.println("5. Agregar nota");
            System.out.println("6. Marcar tarea completada");
            System.out.println("7. Ver estadisticas");
            System.out.println("8. Salir");
            System.out.println("=================================");
            System.out.print("Seleccione una opcion: ");
            opcion = teclado.nextInt();
            teclado.nextLine();
            switch (opcion) {
                //-------------------------//
                //--|ver_horario_por_dia|--//
                //-------------------------//
                case 1:
                    System.out.println("\n===== HORARIO DEL DIA =====");
                    boolean hayActividades = false;
                    for (hora = 0; hora < 24; hora++) {
                        if (!actividades[hora].equals("")) {
                            hayActividades = true;
                            System.out.println("\n" + hora + ":00");
                            System.out.println("Actividad: " + actividades[hora]);
                            System.out.println("Nota: " + notas[hora]);
                            if (completadas[hora]) {
                                System.out.println("Estado: COMPLETADA");
                            } else {
                                System.out.println("Estado: PENDIENTE");
                            }
                            System.out.println("-------------------------");
                        }
                    }
                    if (!hayActividades) {
                        System.out.println("No hay actividades registradas.");
                    }
                    break;
                //----------------------------//
                //--|ver_horario_por_semana|--//
                //----------------------------//
                case 2:
                    System.out.println("\nFuncion disponible en una futura version.");
                    System.out.println("Actualmente solo existe horario diario.");
                    break;
                //-------------------------//
                //--|ver_horario_por_mes|--//
                //-------------------------//
                case 3:
                    System.out.println("\nFuncion disponible en una futura version.");
                    System.out.println("Actualmente solo existe horario diario.");
                    break;
                //-------------------------//
                //--|registrar_actividad|--//
                //-------------------------//
                case 4:
                    System.out.print("\nIngrese la hora (0-23): ");
                    hora = teclado.nextInt();
                    teclado.nextLine();
                    if (hora >= 0 && hora <= 23) {
                        System.out.print("Ingrese la actividad: ");
                        actividad = teclado.nextLine();
                        System.out.print("Ingrese una nota: ");
                        nota = teclado.nextLine();
                        if (actividades[hora].equals("")) {
                            total_actividades++;
                        }
                        actividades[hora] = actividad;
                        notas[hora] = nota;
                        completadas[hora] = false;
                        System.out.println("\nActividad registrada correctamente.");
                    } else {
                        System.out.println("Hora invalida.");
                    }
                    break;
                //------------------------------//
                //--|agregar_o_modificar_nota|--//
                //------------------------------//
                case 5:
                    System.out.print("\nIngrese la hora de la actividad: ");
                    hora = teclado.nextInt();
                    teclado.nextLine();
                    if (hora >= 0 && hora <= 23 &&
                            !actividades[hora].equals("")) {
                        System.out.print("Ingrese la nueva nota: ");
                        nota = teclado.nextLine();
                        notas[hora] = nota;
                        System.out.println("Nota actualizada.");
                    } else {
                        System.out.println("No existe actividad en esa hora.");
                    }
                    break;
                //-----------------------------//
                //--|marcar_tarea_completada|--//
                //-----------------------------//
                case 6:
                    System.out.print("\nIngrese la hora de la actividad: ");
                    hora = teclado.nextInt();
                    teclado.nextLine();
                    if (hora >= 0 && hora <= 23 &&
                            !actividades[hora].equals("")) {
                        completadas[hora] = true;
                        System.out.println(
                                "Actividad completada correctamente.");
                    } else {
                        System.out.println(
                                "No existe actividad registrada.");
                    }
                    break;
                //------------------//
                //--|estadisticas|--//
                //------------------//
                case 7:
                    actividades_completadas = 0;
                    actividades_pendientes = 0;
                    for (hora = 0; hora < 24; hora++) {
                        if (!actividades[hora].equals("")) {
                            if (completadas[hora]) {
                                actividades_completadas++;
                            } else {
                                actividades_pendientes++;
                            }
                        }
                    }
                    System.out.println("\n===== ESTADISTICAS =====");
                    System.out.println(
                            "Total actividades: "
                                    + total_actividades);
                    System.out.println(
                            "Completadas: "
                                    + actividades_completadas);
                    System.out.println(
                            "Pendientes: "
                                    + actividades_pendientes);
                    break;
                //------------------------------//
                //--|salir_del_menu_principal|--//
                //------------------------------//
                case 8:
                    System.out.println(
                            "\nSaliendo del sistema...");
                    teclado.close();
                    return;
                default:
                    System.out.println(
                            "\nOpcion no valida.");
            }
        }
    }
}

File: bloc_de_horarios.py
opcion = 0
hora = 0
actividad = ""
nota = ""
total_actividades = 0
actividades_completadas = 0
actividades_pendientes = 0
actividades = [""] * 24
notas = [""] * 24
completadas = [False] * 24
#-------------------------------------#
#--|menu_principal_bloc_de_horarios|--#
#-------------------------------------#
while True:
    print("=================================")
    print("        BLOC DE HORARIOS")
    print("=================================")
    print("1. Ver horario por dia")
    print("2. Ver horario por semana")
    print("3. Ver horario por mes")
    print("4. Registrar actividad")
    print("5. Agregar nota")
    print("6. Marcar tarea completada")
    print("7. Ver estadisticas")
    print("8. Salir")
    print("=================================")
    opcion = int(input("Seleccione una opcion: "))
    #-------------------------#
    #--|ver_horario_por_dia|--#
    #-------------------------#
    if opcion == 1:
        print("\n===== HORARIO DEL DIA =====")
        hay_actividades = False
        for hora in range(24):
            if actividades[hora] != "":
                hay_actividades = True
                print(f"\n{hora}:00")
                print(f"Actividad: {actividades[hora]}")
                print(f"Nota: {notas[hora]}")
                if completadas[hora]:
                    print("Estado: COMPLETADA")
                else:
                    print("Estado: PENDIENTE")
                print("-------------------------")
        if not hay_actividades:
            print("No hay actividades registradas.")
    #----------------------------#
    #--|ver_horario_por_semana|--#
    #----------------------------#
    elif opcion == 2:
        print("\nFuncion disponible en una futura version.")
        print("Actualmente solo existe horario diario.")
    # ------------------------- #
    # --|ver_horario_por_mes|-- #
    # ------------------------- #
    elif opcion == 3:
        print("\nFuncion disponible en una futura version.")
        print("Actualmente solo existe horario diario.")
    #-------------------------#
    #--|registrar_actividad|--#
    #-------------------------#
    elif opcion == 4:
        try:
            hora = int(input("\nIngrese la hora (0-23): "))
            if 0 <= hora <= 23:
                actividad = input("Ingrese la actividad: ")
                nota = input("Ingrese una nota: ")
                if actividades[hora] == "":
                    total_actividades += 1
                actividades[hora] = actividad
                notas[hora] = nota
                completadas[hora] = False
                print("\nActividad registrada correctamente.")
            else:
                print("Hora invalida.")
        except ValueError:
            print("Debe ingresar un numero.")
    #------------------------------#
    #--|agregar_o_modificar_nota|--#
    #------------------------------#
    elif opcion == 5:
        try:
            hora = int(input("\nIngrese la hora de la actividad: "))
            if 0 <= hora <= 23 and actividades[hora] != "":
                nota = input("Ingrese la nueva nota: ")
                notas[hora] = nota
                print("Nota actualizada.")
            else:
                print("No existe actividad en esa hora.")
        except ValueError:
            print("Debe ingresar un numero.")
    #-----------------------------#
    #--|marcar_tarea_completada|--#
    #-----------------------------#
    elif opcion == 6:
        try:
            hora = int(input("\nIngrese la hora de la actividad: "))
            if 0 <= hora <= 23 and actividades[hora] != "":
                completadas[hora] = True
                print("Actividad completada correctamente.")
            else:
                print("No existe actividad registrada.")
        except ValueError:
            print("Debe ingresar un numero.")
    #------------------#
    #--|estadisticas|--#
    #------------------#
    elif opcion == 7:
        actividades_completadas = 0
        actividades_pendientes = 0
        for hora in range(24):
            if actividades[hora] != "":
                if completadas[hora]:
                    actividades_completadas += 1
                else:
                    actividades_pendientes += 1
        print("\n===== ESTADISTICAS =====")
        print(f"Total actividades: {total_actividades}")
        print(f"Completadas: {actividades_completadas}")
        print(f"Pendientes: {actividades_pendientes}")
    #------------------------------#
    #--|salir_del_menu_principal|--#
    #------------------------------#
    elif opcion == 8:
        print("\nSaliendo del sistema...")
        break
    else:
        print("\nOpcion no valida.")